libdevcore/SHA3.cpp

```diff
@@ -188,54 +174,18 @@ static inline int hash(uint8_t* out, size_t outlen,
 	return 0;
 }
 
-/*** Helper macros to define SHA3 and SHAKE instances. ***/
-#define defshake(bits)                                            \
-  int shake##bits(uint8_t* out, size_t outlen,                    \
-				  const uint8_t* in, size_t inlen) {              \
-	return hash(out, outlen, in, inlen, 200 - (bits / 4), 0x1f);  \
-  }
-#define defsha3(bits)                                             \
-  int sha3_##bits(uint8_t* out, size_t outlen,                    \
-				  const uint8_t* in, size_t inlen) {              \
-	if (outlen > (bits/8)) {                                      \
-	  return -1;                                                  \
-	}                                                             \
-	return hash(out, outlen, in, inlen, 200 - (bits / 4), 0x06);  \
-  }
-#define defkeccak(bits)                                             \
-  int keccak##bits(uint8_t* out, size_t outlen,                    \
-				  const uint8_t* in, size_t inlen) {              \
-	if (outlen > (bits/8)) {                                      \
-	  return -1;                                                  \
-	}                                                             \
-	return hash(out, outlen, in, inlen, 200 - (bits / 4), 0x01);  \
-  }
-
-/*** FIPS202 SHAKE VOFs ***/
-defshake(128)
-defshake(256)
-
-/*** FIPS202 SHA3 FOFs ***/
-defsha3(224)
-defsha3(256)
-defsha3(384)
-defsha3(512)
-
-/*** KECCAK FOFs ***/
-defkeccak(224)
-defkeccak(256)
-defkeccak(384)
-defkeccak(512)
+int keccak256(uint8_t* out, uint8_t const* in, size_t inlen)
+{
+	return hash(out, 32, in, inlen, 200 - (256 / 4), 0x01);
+}
 
 }
 
 bool keccak256(bytesConstRef _input, bytesRef o_output)
 {
-	// FIXME: What with unaligned memory?
 	if (o_output.size() != 32)
 		return false;
```

Comment:
Are callers of this actually checking the return value? If not, should this just be turned into an assertion?